### PR TITLE
Use PhysicalModel for the default canvas Material.

### PR DESCRIPTION
**Pull request by amirh**

### CODE REVIEW

**Comment by amirh** [comment]
It might be easier to review the diff for each commit separately.
The first commit is just a revert of the previous RRect workaround.

File: packages/flutter/test/rendering/recording_canvas.dart [hunk 1]
```diff
@@ -110,14 +110,6 @@ class TestRecordingPaintingContext implements PaintingContext {
...
   }
 
-  @override
-  void pushClipRRect(bool needsCompositing, Offset offset, Rect bounds, RRect clipRRect, PaintingContextCallback painter) {
-    canvas.save();
-    canvas.clipRRect(clipRRect.shift(offset));
-    painter(this, offset);
-    canvas.restore();
-  }
-
```
Comment (Hixie) [inline packages/flutter/test/rendering/recording_canvas.dart]: you can leave this one

**Comment by Hixie** [comment]
LGTM

File: packages/flutter/test/rendering/recording_canvas.dart (hunk 1 above)
Comment (amirh) [reply packages/flutter/test/rendering/recording_canvas.dart]: done

**Comment by amirh** [comment]
@HansMuller - I updated buttons_test to expect clipPath instead of clipRect, please take a look at the last commit in this PR

**Comment by HansMuller** [comment]
LGTM too. Sorry to have taken so long to chime in.

**Pull request merged**